learn/models/conformal-regression/index.qmd

```diff
@@ -0,0 +1,437 @@
+
+To demonstrate, we can take the previous simulation system and induce a variance that is dynamic over the predictor range: 
+
+```{r}
+#| label: variable-plot
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 4
+
+make_variable_data <- function(n, std_dev = 1 / 5) {
+  tibble(x = runif(n, min = -1)) %>%
+    mutate(
+      y = (x^3) + 2 * exp(-6 * (x - 0.3)^2),
+      y = y + rnorm(n, sd = std_dev * abs(x))
+    )
+}
+
+make_variable_data(1000) %>% 
+  ggplot(aes(x, y)) + 
+  geom_point(alpha = 1 / 5)
+```
+
+Let's create new data sets and re-fit our model: 
+
+```{r}
+#| label: variable-data
+set.seed(7292)
+train_variable_data <- make_variable_data(n)
+cal_variable_data <- make_variable_data(250)
+test_variable_data <- make_variable_data(10000)
+
+nnet_variable_fit <- nnet_wflow %>% fit(train_variable_data)
+nnet_variable_pred <- augment(nnet_variable_fit, train_variable_data)
+```
+
+We can recreate the CV+ interval for this new version of the data:
+
+```{r}
+#| label: cv-redux
+#| out-width: 70%
+#| echo: false
+#| fig-width: 6
+#| fig-height: 4 
+set.seed(493)
+folds_variable <- vfold_cv(train_variable_data)
+
+nnet_variable_rs <- 
+  nnet_wflow %>% 
+  fit_resamples(folds_variable, control = ctrl)
+
+cv_variable_int <- int_conformal_cv(nnet_variable_rs)
+
+test_cv_variable_res <- 
+  predict(cv_variable_int, test_variable_data, level = 0.90) %>% 
+  bind_cols(test_variable_data)
+
+test_cv_variable_res %>% 
+  ggplot(aes(x)) +
+  geom_point(aes(y = y), alpha = 1 / 20) +
+  geom_line(
+    data = nnet_variable_pred, aes(y = .pred),
+    linewidth = 1, col = "blue"
+  ) +
+  geom_ribbon(aes(ymin = .pred_lower, ymax = .pred_upper), 
+              col = "#D95F02", linewidth = 1, fill = NA)
+```
+
+While the coverage is good (`r round(coverage(test_cv_variable_res)$coverage, 1)`%), there are obvious areas where it is either too wide or too narrow. 
+
+Conformalized quantile regression ([Romano _et al_ (2019)](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=%22Conformalized+quantile+regression%22)) is a method to produce intervals that can properly scale the intervals based on what was observed in the training data. It fits an initial [quantile regression](https://en.wikipedia.org/wiki/Quantile_regression) model to do so and also required a split data set, such as our calibration data. 
+
+The original work used basic quantile regression models. For tidymodels, we'll use [quantile regression forests](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=%22Quantile+Regression+Forests%22). The reason being that some models require nonlinear terms, interactions, and other features to model the data adequately. Users would have to have a parallel modeling process for that model. Random forests for quantile regression are very adaptable, low maintenance, and are robust to tuning parameter specification. There is a minor downside, as seen below. 
```

Comment:
I really like this. It might be worth it to be a little bit more explicit about that that minor downside is, i.e. refer back to this sentence when you talk about the bumpiness of the intervals?